lib/ansible/modules/cloud/docker/docker_node_facts.py

```diff
@@ -86,22 +106,38 @@
     pass
 
 
-def get_node_facts(client, name):
-    try:
-        return client.inspect_node(name)
-    except NotFound:
-        return None
-    except APIError as exc:
-        if exc.status_code == 503:
-            client.fail(msg="Cannot inspect node: To inspect node execute module on Swarm Manager")
-        client.fail(msg="Error while reading from Swarm manager: %s" % to_native(exc))
-    except Exception as exc:
-        client.module.fail_json(msg="Error inspecting swarm node: %s" % exc)
+def get_node_facts(client):
+
+    results = []
+
+    if client.module.params['self'] is True:
+        try:
+            self_node_id = client.get_swarm_node_id()
+        except APIError:
+            return None
```

Comment:
I think it makes sense to fail here instead. (Or return `[]`.)

Reply:
Actually, it may be removed as a whole - at some point, the APIError handling has been moved to `get_swarm_node_id()` itself in `module_utils\docker_swarm.py` library.